concent_api/sci_testing_common.py

```diff
@@ -0,0 +1,261 @@
+    @staticmethod
+    def _test_gnt_to_gntb_transfer(sci: SmartContractsInterface) -> None:
+        sci.transfer_gnt(
+            sci.get_gate_address(),
+            sci.get_gnt_balance(sci.get_eth_address())
+        )
+        sci.transfer_from_gate()
+
+    def _wait_unitl_timeout(self, condition: Callable, timeout_message: str, sleep_message: str) -> None:
+
+        start = time.time()
+
+        while condition():
+            print(sleep_message)
+            time.sleep(self.sleep_time)
+            if start + self.timeout < time.time():
+                raise TimeoutError(timeout_message)
+        else:
+            print(f'Sleep lasted for {int(time.time() - start)} seconds. Transaction confirmed.')
+
+    def request_for_gntb(self) -> None:
+        self._test_eth_faucet_donate(self.requestor_sci.get_eth_address())
+        self._test_eth_faucet_donate(self.provider_sci.get_eth_address())
+
+        self._wait_unitl_timeout(
+            condition=lambda: (
+                self.requestor_sci.get_eth_balance(self.requestor_sci.get_eth_address()) == 0 or
+                self.provider_sci.get_eth_balance(self.provider_sci.get_eth_address()) == 0
+            ),
+            timeout_message='Test ETH faucet timeout',
+            sleep_message='Waiting for test ETH for Provider and Requestor...',
+        )
+
+        self._test_gnt_faucet(self.requestor_sci)
+        self._test_gnt_faucet(self.provider_sci)
+
+        self._wait_unitl_timeout(
+            condition=lambda: (
+                self.requestor_sci.get_gnt_balance(self.requestor_sci.get_eth_address()) == 0 or
+                self.provider_sci.get_gnt_balance(self.provider_sci.get_eth_address()) == 0 or
+                self.requestor_sci.get_gate_address() is None or
+                self.provider_sci.get_gate_address() is None
+            ),
+            timeout_message='Test GNT faucet timeout',
+            sleep_message='Waiting for test GNT for Provider and Requestor...',
+        )
+
+        self._test_gnt_to_gntb_transfer(self.requestor_sci)
+        self._test_gnt_to_gntb_transfer(self.provider_sci)
+
+        self._wait_unitl_timeout(
+            condition=lambda: (
+                self.requestor_sci.get_gntb_balance(self.requestor_sci.get_eth_address()) == 0 or
+                self.provider_sci.get_gntb_balance(self.provider_sci.get_eth_address()) == 0
+            ),
+            timeout_message='Test GNT to GNTB transfer timeout',
+            sleep_message='Waiting for test GNTB for Provider and Requestor...',
+        )
+
+    def request_for_deposit(self, value: int=0):
+        if value < 0:
+            raise ValueNotPositiveError('Value must be greater than zero')
+        requestor_gntb_balance = self.requestor_sci.get_gntb_balance(self.requestor_sci.get_eth_address())
+        provider_gntb_balance = self.provider_sci.get_gntb_balance(self.provider_sci.get_eth_address())
+        if requestor_gntb_balance == 0 or value > requestor_gntb_balance:
```

Comment:
Please reverse the `<` condition.

Reply:
I am not sure what did you mean because condition itself is good. Do you want me to change it to: ``` requestor_gntb_balance < value``` ?

Reply:
Yes, exactly.

Reply:
Ok, Done :)